infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/types/CommonTypeDefinitions.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.restapi.types;
+
+import java.util.Optional;
+import java.util.function.Function;
+import tech.pegasys.teku.infrastructure.http.HttpErrorResponse;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+
+public class CommonTypeDefinitions {
+  public static final DeserializableTypeDefinition<String> STRING_TYPE =
+      new StringBasedPrimitiveTypeDefinition<>(
+          Function.identity(), Function.identity(), Optional.empty(), Optional.empty());
```

Comment:
maybe a constructor that doesnt take the final two optionals? or are we wanting to discourage leaving them empty?

Reply:
Yeah probably a good idea.  I'm thinking that `StringBasedPrimitiveTypeDefinition` would only be created here anyway.  Made the constructors package private to reinforce that.